src/NGen.cpp

```diff
@@ -327,74 +331,99 @@ int main(int argc, char *argv[]) {
 
     std::shared_ptr<pdm03_struct> pdm_et_data = std::make_shared<pdm03_struct>(get_et_params());
 
+    // check the time loops for the existing layers
+    ngen::LayerDataStorage& layer_meta_data = manager->get_layer_metadata();
+
+    // get the keys for the existing layers
+    std::vector<int>& keys = layer_meta_data.get_keys();
+
+    // get the converted time steps for layers
+    double min_time_step;
+    double max_time_step;
+    std::vector<double> time_steps;
```

Comment:
```suggestion
    std::vector<double> time_steps;
    time_steps.reserve(keys.size());
```
This will allocate the memory in one go, and avoid initializing it just to then overwrite it below